fusion-engine-driver/transport/data_listener.hpp

```diff
@@ -1,14 +1,65 @@
 #pragma once
 
-#include <functional>
 #include <cstdint>
+#include <functional>
+
+/**
+ * @brief An abstract base class that defines a data listener interface.
+ *
+ * The DataListener class is used as a base class for implementing different
+ * types of data listeners, such as a TCP listener or a serial listener. It
+ * defines a set of pure virtual functions that must be implemented by any
+ * subclass, including functions for starting and stopping the listener, writing
+ * data to the listener, and setting a callback function for handling incoming
+ * data.
+ */
+class DataListener {
+ public:
+  virtual ~DataListener() = default;
+
+  /**
+   * @brief Starts the data listener.
+   *
+   * This pure virtual function must be implemented by any subclass of
+   * DataListener. It is responsible for starting the data listener and
+   * listening for incoming data.
+   */
+  virtual void listen() = 0;
+
+  /**
+   * @brief Stops the data listener.
+   *
+   * This virtual function can be overridden by any subclass of DataListener to
+   * provide additional functionality for stopping the data listener. By
+   * default, it simply sets the `running_` flag to false.
+   */
+  virtual void stop() { running_ = false; }
+
+  /**
+   * @brief Writes data to the data listener.
+   *
+   * This pure virtual function must be implemented by any subclass of
+   * DataListener. It is responsible for writing data to the data listener,
```

Comment:
It is responsible for writing data to the underlying socket protocole (e.g. serial or TCP), in our case, mainly to send RTCM corrections to our quectel device in serial mode.